### PR TITLE
More test cases

### PR DESCRIPTION
**Pull request by mkrajcov**
infinispan, deployment scanners, ee subsystem, metrics

### CODE REVIEW

**Comment by heiko-braun** [comment]
Any update on the HAL version you are using? it should be >=2.7.x since overall navigation has changed pretty much ...

**Comment by heiko-braun** [comment]
i mention this because navigating the url fragment might not work in all cases any more

**Comment by zebrik** [comment]
Unfortunately this tests are developed against 2.6.x still.

File: src/test/java/org/jboss/hal/testsuite/test/runtime/DataSourceMetricsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    private WebDriver browser;
+
+    @Page
+    private DataSourcesMetricsPage dsPage;
+
+    @Before
+    public void before(){
+        browser.navigate().refresh();
+        Graphene.goTo(DataSourcesMetricsPage.class);
+        browser.manage().window().maximize();
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/runtime/DataSourceMetricsTestCase.java]: Did you consider to use org.jboss.hal.testsuite.util.Console.maximizeWindow() instead?

File: src/test/java/org/jboss/hal/testsuite/test/runtime/JvmMetricsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private WebDriver browser;
+
+    @Page
+    private JVMPage jvmPage;
+
+    @Before
+    public void before(){
+        browser.navigate().refresh();
+        Graphene.goTo(JVMPage.class);
+        browser.manage().window().maximize();
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/runtime/JvmMetricsTestCase.java]: Did you consider to use org.jboss.hal.testsuite.util.Console.maximizeWindow() instead?

File: src/test/java/org/jboss/hal/testsuite/test/runtime/TransactionsMetricsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    private WebDriver browser;
+
+    @Page
+    private TransactionsMetricsPage tmPage;
+
+    @Before
+    public void before(){
+        browser.navigate().refresh();
+        Graphene.goTo(TransactionsMetricsPage.class);
+        browser.manage().window().maximize();
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/runtime/TransactionsMetricsTestCase.java]: Did you consider to use org.jboss.hal.testsuite.util.Console.maximizeWindow() instead?

File: src/test/java/org/jboss/hal/testsuite/test/runtime/WebServiceEndpointsTestCase.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private WebDriver browser;
+
+    @Page
+    private WebServiceEndpointsPage wsePage;
+
+    @Before
+    public void before(){
+        browser.navigate().refresh();
+        Graphene.goTo(WebServiceEndpointsPage.class);
+        browser.manage().window().maximize();
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/runtime/WebServiceEndpointsTestCase.java]: Did you consider to use org.jboss.hal.testsuite.util.Console.maximizeWindow() instead?

File: src/test/java/org/jboss/hal/testsuite/test/configuration/container/EETestCase.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Page
+    public EEPage page;
+
+    @Before
+    public void before(){
+        browser.navigate().refresh();
+        Graphene.goTo(EEPage.class);
+        Console.withBrowser(browser).waitUntilLoaded();
+        Console.withBrowser(browser).maximizeWindow();
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/configuration/container/EETestCase.java:43]: Can be chained ;-) ... in more places of this PR

File: src/test/java/org/jboss/hal/testsuite/test/configuration/infinispan/DistributedCacheTestCase.java [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        checker.editTextAndAssert(page, "stopTimeout", "-15000").expectError()
+                .tab("Trans").rowName(CACHE_NAME).defineFirst("hasTransaction").invoke();
+    }
+
+    @Test
+    @InSequence(19)
+    public void removeCache() {
+        page.content().getResourceManager().removeResourceAndConfirm(CACHE_NAME);
+    }
+}
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/configuration/infinispan/DistributedCacheTestCase.java:258]: @InSequence(1)...@InSequence(19) Wow! Is such test dependency sequence really necessary or could it be redesigned?

File: src/test/java/org/jboss/hal/testsuite/test/configuration/infinispan/InvalidationCacheTestCase.java [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+        checker.editTextAndAssert(page, "stopTimeout", "-15000").expectError()
+                .tab("Trans").rowName(CACHE_NAME).defineFirst("hasTransaction").invoke();
+    }
+
+    @Test
+    @InSequence(19)
+    public void removeCache() {
+        page.content().getResourceManager().removeResourceAndConfirm(CACHE_NAME);
+    }
+}
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/configuration/infinispan/InvalidationCacheTestCase.java:258]: The same question about test dependency necessity as for DistributedCacheTestCase.

File: src/test/java/org/jboss/hal/testsuite/test/util/ConfigAreaChecker.java [hunk 8]
```diff
@@ -145,16 +182,19 @@ public void invoke() {
             }
             fragment.saveAndAssert(expectedChange);
```
Comment (zebrik) [inline src/test/java/org/jboss/hal/testsuite/test/util/ConfigAreaChecker.java]: I know that this controversy line is not caused by this PR but I belive that we should avoid and deprecate usage of asserts in UI elements. Mixture of UI and assert logic is neither clean nor safe (hiding failure reasons by Graphene proxies.) Could you please use 

```
fragment.save()
```

instead and extract assert logic here?

**Comment by mkrajcov** [comment]
updated after suggestions

**Comment by zebrik** [comment]
Closing this pull request since Michal is currently rewriting these and even newer tests for HAL 2.8

**Comment by heiko-braun** [comment]
I think the majority of the tests (the actual subsystem screens) should still be usable.

**Pull request closed without merging**